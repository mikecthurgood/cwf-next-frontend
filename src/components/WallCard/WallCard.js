import React from 'react';
import StarRatings from 'react-star-ratings';
import Link from 'next/link';
import Styled from 'styled-components';

const WallCard = ({ wall }) => (
    <Card>
        <Link href={`/walls/${wall.slug}`}><a>
            <Heading>
                <h5>{wall.name}</h5>
                {wall.distance && <h4>({wall.distance} miles)</h4>}
            </Heading>
            <ImageContainer>
                <picture className='wall-image'>
                    <source srcSet={wall.imageUrl} type='image/webp'></source>
                    <source srcSet={wall.imageUrl.replace('webp', 'jpg').replace('/climbing-walls', '/climbing-walls/jpegs')} type='image/jpeg'></source>
                    <img src={wall.imageUrl.replace('webp', 'jpg').replace('/climbing-walls', '/climbing-walls/jpegs')} className='wall-image' alt="" />
                </picture>
            </ImageContainer>
            <Content>
                <div className='wall-details-1'>
                    <p><strong>Location:</strong> {wall.region}</p>
                    <><strong>User Rating:</strong><br />
                        {wall.reviews && wall.reviews.length > 0 ?
                            <StarRatings
                                rating={wall.reviews.reduce((prev, cur) => { return prev + cur.rating }, 0) / wall.reviews.length}
                                starRatedColor="gold"
                                numberOfStars={5}
                                name='rating'
                                starDimension="20px"
                                starSpacing="0px"
                            />
                            :
                            'No ratings yet'}</>
                    <strong>{wall.boulderingOnly && <p style={{ color: 'red' }}>Bouldering Only</p>}</strong>
                </div>
                <div className='wall-details-2'>

                    <p><strong>Weekdays:</strong><br />
                        Open: {wall.weekdayOpening}<br />
                        Close: {wall.weekdayClosing}</p>
                    <p><strong>Weekends:</strong><br />
                        Open: {wall.weekendOpening}<br />Close: {wall.weekendClosing}</p>
                </div>
            </Content>
            <Btn>View Details and Reviews</Btn>
        </a></Link>
    </Card>
)

export default WallCard

const Card = Styled.div`
    flex: 1;
    flex-grow: 1;
    flex-basis: 270px;
    height: 100%;
    min-width: 400px;
    margin: 2px;
    border-radius: 2px;
    border-style: solid;
    border-color: #f5f5f5;
    border-width: 0.04em;
    text-align: center;
    box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
    transition: 0.5s;
    overflow: hidden;
    position: relative;
    background-color: #fff;
    color: #282c34;
    text-decoration: none;
    cursor: default;

    @media only screen and (max-width: 480px) {
      min-width: 100vw;
      border-radius: 0;
    }

    &:hover {
        box-shadow: 0 4px 8px 0 rgba(0,0,0,0.8);
    }
`

const Heading = Styled.div`
    color: #282c34;
    border-width: 0em;
    border-bottom: 0.04em;
    border-color: lightgray;
    border-style: solid;
    padding: 10px 0;
    h4 {
        padding: 0;
        margin: 0;
        color: #ED5656;
    }
    h5 {
        font-size: 16px;
        margin: 0;
        text-decoration: none;

        @media only screen and (max-width: 480px) {
            font-size: 20px;
        }
    }
`

const ImageContainer = Styled.div`
    width: 100%;
    height: 250px;
    
    .wall-image {
        height: 100%;
        min-width: 100%;

        img {
            width: 100%;
            object-fit: cover;
        }
    }
`
const Content = Styled.div`
    text-align: left;
    font-size: 14px;
    width: 100%;  
    position: relative;
    height: 155px;

    @media only screen and (max-width: 480px) {
        font-size: 16px;
    }

    .wall-details-1 {
        width: fit-content;
        position: relative;
        float: left;
        padding-left: 10px; 
        @media only screen and (max-width: 480px) {
            width: 45%;
        }
    }

    .wall-details-2 {
        width: 100px;
        position: relative;
        float: right;
        padding-right: 10px;
        @media only screen and (max-width: 480px) {
            width: 45%;
            text-align: right; 
        }
    }
`

const Btn = Styled.button`
    padding: 5px 30px;
    border-radius: 2px;
    background-color: #ED5656;
    color: white;
    font-size: 14px;
    width: 100%;
    font-family: 'Muli', sans-serif;
    @media only screen and (max-width: 974px) {
        font-size: 12px;
    }
    @media only screen and (max-width: 480px) {
        font-size: 20px;
    }

    &:focus {
        outline:0;
    }

    &:hover {
        background-color: rgb(179, 64, 64);
    }
`