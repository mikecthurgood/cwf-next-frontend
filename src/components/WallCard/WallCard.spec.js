import React from 'react';
import { render } from '@testing-library/react';
import WallCard from '.';
import mockWalls from '../../../__tests__/mockWalls.json';
jest.mock('react-star-ratings');

describe('Wallcard', () => {
    it('should render a wall with ratings correctly', () => {
        const wall = mockWalls.wallsWithDistance[0]
        const { container } = render(
            <WallCard wall={wall} />
        );
        
        expect(container).toMatchSnapshot();
    });

    it('should render a wall without ratings correctly', () => {
        const wall = mockWalls.wallsWithDistance[1]
        const { container } = render(
            <WallCard wall={wall} />
        );
        
        expect(container).toMatchSnapshot();
    });
})