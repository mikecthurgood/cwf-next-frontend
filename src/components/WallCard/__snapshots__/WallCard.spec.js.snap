// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallcard should render a wall with ratings correctly 1`] = `
<div>
  <div
    class="WallCard__Card-sc-1uzo1sy-0 gsbJUd"
  >
    <a
      href="/walls/test"
    >
      <div
        class="WallCard__Heading-sc-1uzo1sy-1 hDrFCI"
      >
        <h5>
          Test Wall 1
        </h5>
        <h4>
          (
          1.5
           miles)
        </h4>
      </div>
      <div
        class="WallCard__ImageContainer-sc-1uzo1sy-2 cEFXNR"
      >
        <picture
          class="wall-image"
        >
          <source
            srcset="/testimageurl"
            type="image/webp"
          />
          <source
            srcset="/testimageurl"
            type="image/jpeg"
          />
          <img
            alt=""
            class="wall-image"
            src="/testimageurl"
          />
        </picture>
      </div>
      <div
        class="WallCard__Content-sc-1uzo1sy-3 evoHFp"
      >
        <div
          class="wall-details-1"
        >
          <p>
            <strong>
              Location:
            </strong>
             
            test
          </p>
          <strong>
            User Rating:
          </strong>
          <br />
          <strong>
            <p
              style="color: red;"
            >
              Bouldering Only
            </p>
          </strong>
        </div>
        <div
          class="wall-details-2"
        >
          <p>
            <strong>
              Weekdays:
            </strong>
            <br />
            Open: 
            10:00
            <br />
            Close: 
            22:00
          </p>
          <p>
            <strong>
              Weekends:
            </strong>
            <br />
            Open: 
            10:00
            <br />
            Close: 
            21:00
          </p>
        </div>
      </div>
      <button
        class="WallCard__Btn-sc-1uzo1sy-4 kMdqoT"
      >
        View Details and Reviews
      </button>
    </a>
  </div>
</div>
`;

exports[`Wallcard should render a wall without ratings correctly 1`] = `
<div>
  <div
    class="WallCard__Card-sc-1uzo1sy-0 gsbJUd"
  >
    <a
      href="/walls/test"
    >
      <div
        class="WallCard__Heading-sc-1uzo1sy-1 hDrFCI"
      >
        <h5>
          Test Wall 2
        </h5>
        <h4>
          (
          1.5
           miles)
        </h4>
      </div>
      <div
        class="WallCard__ImageContainer-sc-1uzo1sy-2 cEFXNR"
      >
        <picture
          class="wall-image"
        >
          <source
            srcset="/testimageurl"
            type="image/webp"
          />
          <source
            srcset="/testimageurl"
            type="image/jpeg"
          />
          <img
            alt=""
            class="wall-image"
            src="/testimageurl"
          />
        </picture>
      </div>
      <div
        class="WallCard__Content-sc-1uzo1sy-3 evoHFp"
      >
        <div
          class="wall-details-1"
        >
          <p>
            <strong>
              Location:
            </strong>
             
            test
          </p>
          <strong>
            User Rating:
          </strong>
          <br />
          No ratings yet
          <strong>
            <p
              style="color: red;"
            >
              Bouldering Only
            </p>
          </strong>
        </div>
        <div
          class="wall-details-2"
        >
          <p>
            <strong>
              Weekdays:
            </strong>
            <br />
            Open: 
            10:00
            <br />
            Close: 
            22:00
          </p>
          <p>
            <strong>
              Weekends:
            </strong>
            <br />
            Open: 
            10:00
            <br />
            Close: 
            21:00
          </p>
        </div>
      </div>
      <button
        class="WallCard__Btn-sc-1uzo1sy-4 kMdqoT"
      >
        View Details and Reviews
      </button>
    </a>
  </div>
</div>
`;
